src/mudlet.cpp

```diff
@@ -2354,6 +2351,16 @@ void mudlet::readEarlySettings(const QSettings& settings)
     // as soon as possible as well!
 
     mShowIconsOnMenuCheckedState = static_cast<Qt::CheckState>(settings.value("showIconsInMenus", QVariant(Qt::PartiallyChecked)).toInt());
+
+    // TODO: At some point (between mudlet 4.0 and 4.1 perhaps) removal this "if" and only consider the QSetting - drop considering the sentinel file:
```

Comment:
I don't see a need to remove this backwards compatibility in Mudlet 4.x. It's very trivial to keep around.

Reply:
It is messy to rely on the presence/absence of an empty file when we can quickly and easily use another setting in the main `QSetting`s system instead - which is ideally suited to maintain this sort of configuration setting (especially if we are trying to provide a portable system/setup where this file presence/absence is *another* detail that #1695 will have to handle) - I was commenting on what I thought would be a reasonable point to abandon considering/maintaining the file altogether i.e. the first minor level change after a major semantic version change.

Reply:
Still no need to include inconvenience users. We can keep this around, it's not a problem to have a fallback.

Reply:
With a **semantic version scheme** such as we have adopted it seems unnecessarily complicated to go about creating/destroying a file just so that versions prior to (say) 3.13 know whether to provided a toolbar button to exit/enter full-screen mode and to start in full-screen mode every time Mudlet is started beyond, say, version **4.1**.

Maintaining that method when it can be better centralised with other application wide settings as part of I would like to do with this PR makes sense as a major version number change detail: i.e. versions 3.13 to 4.0 will use the `QSetting` based system but also make sure that the file is present or absent to keep previous versions in step, however 4.1 would not and then could have simpler code.  The only difference the end user would see is that if they ran a version prior to 3.13 AND a version later than 4.0 is that should one version change whether it is to have the full-screen option enabled then this would no longer carry over into the other automatically and then the user would have to make the change in the setting manually.  By planning to run both methods between 3.13 and 4.0 we can ensure a seamless and orderly change over and eventually make a code simplification without 99% (I expect) of people noticing and hardly inconveniencing the remainder.  The benefit is that we can eliminate one atypically stored application detail and move it into the same system that others are using - and thus make the code to handle it more like that for other features/options...

IMVHO It is a problem to keep the fall-back for an *indefinite* time because it will hinder, for example: developments in making the application *portable* because it is an additional detail to handle/maintain/debug/test.  **Remember that this detail has come up in *this* PR because it is an application wide setting that had to be kept in step between multiple profile preference dialogues and that creating/destroying the empty file that signalled the state of this feature would get rather complex with multiple instances of the `dlgProfilePrefence` class each wanting to do this as the original code did!** :open_mouth: 

I am not saying that the fall-back is not desirable now and going forward - I am just saying that it should not be maintained forever and that the indicated version numbers makes sense *to me* as a reasonable life-span for keeping it in play.

Reply:
Okay - but I disagree and I'd rather not have comments declaring that on so and so version, we'll break things just slightly for the 1% of the usecases po it there. It's just unnecessary, we can live with it. We did not adopt semantic versioning so we can have an excuse to break stuff.

I don't think it'll hinder Mudlet going portable at all as it's not an issue for the current PR.

Let's take the supposing comments out and if we actually get pressed up against the wall in the future, we'll consider then what to do.

Reply:
I've compromised and taken out all references to *when* such a change might take place but have left identifiers in the two places where the change would be needed as `// PLACEMARKER: Full-screen mode controlled by File (# of 2) ...` which means (if you use the *TODO list Qt IDE plugin as I do) it should be easy to find them if we ever decide to do it.